api/src/test/java/org/bahmni/module/events/publisher/EventPublisherTest.java

```diff
@@ -0,0 +1,68 @@
+package org.bahmni.module.events.publisher;
+
+import org.bahmni.module.events.model.Event;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.openmrs.Person;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.jms.core.MessageCreator;
+
+import java.io.IOException;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EventPublisherTest {
+	
+	@Test
+	public void shouldPublishEventOnTopicNameAsEventType() {
+		EventPublisher eventPublisher;
+		JmsTemplate jmsTemplate = mock(JmsTemplate.class);
+		ObjectMapper objectMapper = new ObjectMapper();
+		eventPublisher = new EventPublisher(jmsTemplate, objectMapper);
+
+		Person person = getPerson();
+		String eventType = "bahmni-patient";
```

Comment:
Strings everywhere : ) Dangerous. https://refactoring.guru/smells/primitive-obsession

Reply:
Used enum instead of string